### PR TITLE
Add tests, bump nodejs to 18

### PR DESCRIPTION
**Pull request by legobeat**
Adds tests for the recently added functionality in #41.

nodejs and babel upgrades required to make jest and coverage working.

The single failing test is addressed in #48 .

### CODE REVIEW

**Comment by legobeat** [comment]
The failing test is addressed in #48

**Comment by legobeat** [comment]
Update: abe25adcd4efb4c2bd9df9864e1200df9439e9eb adjusts regexes and adds one previously failing case.

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  // Automatically clear mock calls, instances and results before every test.
+  // This does not remove any mock implementation that may have been provided,
+  // so we disable it.
+  // clearMocks: true,
+
+  // Indicates whether the coverage information should be collected while executing the test
+  collectCoverage: true,
+
+  // An array of glob patterns indicating a set of files for which coverage information should be collected
+  collectCoverageFrom: ['./src/**/*.ts'],
```
Comment (mcmire) [inline jest.config.js:25]: Should this be `app/**/*.js`?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  // Indicates which provider should be used to instrument code for coverage
+  coverageProvider: 'babel',
+
+  // A list of reporter names that Jest uses when writing coverage reports
+  coverageReporters: ['html', 'json-summary', 'text'],
+
+  // An object that configures minimum threshold enforcement for coverage results
+  coverageThreshold: {
+    global: {
+      branches: 100,
```
Comment (mcmire) [inline jest.config.js:44]: Might need to intentionally lower these thresholds for now.

File: app/lib.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+const fs = require('fs')
+const path = require('path')
+const {
+  decryptVault,
+  extractVaultFromFile,
+} = require('./lib.js')
+
+const crypto = require('crypto');
+
+Object.defineProperty(globalThis, 'crypto', {
```
Comment (mcmire) [inline app/lib.test.js:10]: Instead of modifying the `crypto` module directly, can we use `jest.mock` to mock it out?

File: app/lib.test.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  {
+    path: 'chromium-94.0.4606.81_4.17/000003.log',
+    mnemonic: 'very follow angry proof column rail smile intact broom chicken lens earth',
+    passphrase: 'aePaf7aequukoo6lahraitheemu6pein',
+  },
+  {
+    path: 'chromium-90-0.4430.72_2.14.1/Local Storage/leveldb/000003.log',
+    mnemonic: 'speed accuse odor ordinary exercise truly outer mask arrest life sibling height',
+    //passphrase: 'bG82kXdp3rNwJ3MCT3kLmLKFN',
```
Comment (mcmire) [inline app/lib.test.js:35]: Any reason why this is commented out?

Comment (legobeat) [reply app/lib.test.js:35]: Yes: It's not actually used (as the SRP is actually stored in clear) but left it in should it ever be relevant to get the passphrase for this wallet.
Might as well add a comment to that effect here.

File: jest.config.js (hunk 1 above)
Comment (legobeat) [reply jest.config.js:25]: Indeed!

File: app/lib.test.js (hunk 3 above)
Comment (legobeat) [reply app/lib.test.js]: I was not able to make it work using `jest.mock` - and I also think that mocking is not correct here as we still want to use the runtime module as-is.

I went with moving this part out, as suggested by [jest docs](https://jestjs.io/docs/configuration#globals-object).

1821fa532bc600a3546a071a33d665c2d727ea9f

File: jest.config.js (hunk 1 above)
Comment (legobeat) [reply jest.config.js]: Ended up restricting it to `lib.js` - i'm not sure if there are ambitions to consider covering the web ui parts for now.

**Comment by legobeat** [comment]
babel upgrade is now part of this PR. needed in order to get coverage working.

8250d674f70fa61062fc464a9396ee45bdbced4a

**Review by mcmire — APPROVED** [review]
LGTM!

**Pull request merged**